File: packages/database/src/models/MentorshipSession.ts
import mongoose, { Schema, Document } from 'mongoose';
import { IMentorshipSession } from '@rural/types';

export interface IMentorshipSessionDocument extends Omit<IMentorshipSession, 'id'>, Document {}

const sessionNotesSchema = new Schema({
  mentorObservations: { type: String },
  businessChallenges: { type: String },
  recommendations: { type: String },
  nextActions: { type: String },
  entrepreneurProgress: { type: String }
}, { _id: false });

const sessionSchema = new Schema<IMentorshipSessionDocument>({
  mentorId: { type: String, required: true, index: true },
  businessId: { type: String, required: true, index: true },
  entrepreneurId: { type: String, required: true, index: true },
  status: { type: String, enum: ['Requested', 'Pending', 'Accepted', 'Rejected', 'Scheduled', 'In Progress', 'Completed', 'Cancelled', 'No Show', 'Rescheduled'], default: 'Requested', index: true },
  scheduledAt: { type: Date },
  durationMinutes: { type: Number, required: true },
  meetingPlatform: { type: String, enum: ['Google Meet', 'Zoom', 'Microsoft Teams', 'In-App WebRTC'], default: 'Google Meet' },
  meetingLink: { type: String },
  goals: { type: [String], default: [] },
  notes: sessionNotesSchema,
  rating: { type: Number, min: 1, max: 5 },
  feedback: { type: String }
}, { timestamps: true });

export const MentorshipSessionModel = mongoose.models.MentorshipSession || mongoose.model<IMentorshipSessionDocument>('MentorshipSession', sessionSchema);

File: apps/mobile/lib/features/mentorship/domain/entities/mentorship_session.dart
enum MentorshipSessionStatus {
  requested,
  confirmed,
  rejected,
  cancelled,
  rescheduleRequested,
  completed,
  noShow
}

class MentorshipSession {
  final String id;
  final String mentorId;
  final String mentorName;
  final String businessId;
  final String entrepreneurId;
  final MentorshipSessionStatus status;
  final DateTime scheduledAt;
  final int durationMinutes;
  final String meetingPlatform;
  final String? meetingLink;
  final List<String> goals;
  final double? rating;
  final String? feedback;
  final bool canJoin;

  const MentorshipSession({
    required this.id,
    required this.mentorId,
    required this.mentorName,
    required this.businessId,
    required this.entrepreneurId,
    required this.status,
    required this.scheduledAt,
    required this.durationMinutes,
    required this.meetingPlatform,
    this.meetingLink,
    required this.goals,
    this.rating,
    this.feedback,
    required this.canJoin,
  });
}
